phpstan.neon.dist

```diff
@@ -29,3 +29,14 @@ parameters:
 			analyseAndScan:
 				- node_modules (?)
 
+		# Ignore specific errors
+		ignoreErrors:
+			# WP_REST_Request is not actually a generic class in WordPress core.
+			# PHPStan's WordPress stubs appear to define it as generic for type checking,
+			# but WordPress itself doesn't use generics. This seems to be an incompatibility
```

Comment:
@justlevine I'm not sure if there is a better way to handle this... but I was having trouble typing this in a way that worked. Any ideas?

Reply:
This issue is that WP_REST_Request expects the generic of args, e.g. `\WP_REST_Request<array:{name:string}>` for example on `WP_REST_Abilities_List_Controller::get_item()`.

If it's not important we can use `WP_REST_Request<array<string,mixed>>`, though it might also go away if we just use `{@inheritDoc}`, but im not by my desk rn to check.

(Either way probably should be {@inheritDoc} in most of those functions).

Might make sense to disable `SlevomatCodingStandard.TypeHints.*.MissingNativeTypeHint` on those classes though since the types are only hinted by the parent, not enforced.

Reply:
I landed this PR without addressing it. Let's figure out the strategy for type hinting seperately. I didn't want this larger PR to get blocked on code quality concerns.

Reply:
See https://github.com/swissspidy/media-experiments/blob/4aace63141fbd57cbece2d574129adfde3d6793d/inc/class-rest-attachments-controller.php for an example on how to properly do this in PHPStan. It's totally doable and there is definitely no need to ignore the errors.

Reply:
@swissspidy the question isnt whether it's doable, its whether it's justifiable when we need to strip out all the non-WPCS syntax in a few months... 🤷

Reply:
It doesn't need to be stripped out though.